File: dev_mock_server.py
import os
import json
from uuid import uuid4
from flask import Flask, request, jsonify, make_response

app = Flask(__name__)

# Simple development storage mock
UPLOAD_DIR = os.path.join(os.getcwd(), "dev_uploads")
os.makedirs(UPLOAD_DIR, exist_ok=True)

def corsify(response, origin=None):
    """Simple CORS helper for development"""
    if origin:
        response.headers["Access-Control-Allow-Origin"] = origin
        response.headers["Access-Control-Allow-Methods"] = "GET,POST,DELETE,OPTIONS,PUT,PATCH"
        response.headers["Access-Control-Allow-Headers"] = "Content-Type,x-api-key,x-tenant-slug,Authorization"
        response.headers["Access-Control-Allow-Credentials"] = "true"
    return response

@app.route("/upload", methods=["POST"])
def mock_upload():
    """Simple mock upload endpoint for development"""
    origin = request.headers.get("Origin")
    
    if 'file' not in request.files:
        return corsify(jsonify({"ok": False, "error": "no_file"}), origin), 400
    
    file = request.files['file']
    if file.filename == '':
        return corsify(jsonify({"ok": False, "error": "no_filename"}), origin), 400
    
    family_slug = request.form.get('family_slug', 'default')
    upload_type = request.form.get('type', 'photo')
    
    # Generate mock response
    file_id = str(uuid4())
    filename = file.filename
    
    # Save file locally for development
    local_path = os.path.join(UPLOAD_DIR, f"{file_id}_{filename}")
    file.save(local_path)
    
    # Get file size
    file_size = os.path.getsize(local_path)
    
    response_data = {
        "ok": True,
        "id": file_id,
        "key": f"dev/{family_slug}/{file_id}/{filename}",
        "type": upload_type,
        "filename": filename,
        "size": file_size,
        "publicUrl": f"http://localhost:5000/uploads/{file_id}_{filename}",
        "message": "File uploaded to local development storage"
    }
    
    print(f"✅ Mock upload successful: {filename} ({file_size} bytes)")
    return corsify(jsonify(response_data), origin)

@app.route("/uploads/<filename>")
def serve_upload(filename):
    """Serve uploaded files for development"""
    return f"File: {filename} (stored locally for development)"

@app.route("/upload", methods=["OPTIONS"])
@app.route("/health", methods=["GET", "OPTIONS"])
@app.route("/status", methods=["GET", "OPTIONS"])
def handle_options():
    origin = request.headers.get("Origin")
    response = make_response(("", 204))
    return corsify(response, origin)

@app.route("/health")
@app.route("/status")
def health():
    origin = request.headers.get("Origin")
    return corsify(jsonify({"status": "ok", "message": "Development mock server"}), origin)

if __name__ == "__main__":
    print("🚀 Starting development mock upload server...")
    print(f"📁 Upload directory: {UPLOAD_DIR}")
    print("🌐 Server running on http://localhost:5000")
    print("📸 Ready to accept uploads!")
    app.run(host="0.0.0.0", port=5000, debug=True)

File: dev_mock_server.js
const express = require('express');
const multer = require('multer');
const path = require('path');
const fs = require('fs');
const { v4: uuidv4 } = require('uuid');

const app = express();
const port = 5000;

// Create uploads directory
const uploadDir = path.join(__dirname, 'dev_uploads');
if (!fs.existsSync(uploadDir)) {
  fs.mkdirSync(uploadDir, { recursive: true });
}

// Configure multer for file uploads
const storage = multer.diskStorage({
  destination: (req, file, cb) => {
    cb(null, uploadDir);
  },
  filename: (req, file, cb) => {
    const fileId = uuidv4();
    cb(null, `${fileId}_${file.originalname}`);
  }
});

const upload = multer({ storage });

// CORS middleware
app.use((req, res, next) => {
  const origin = req.headers.origin;
  if (origin) {
    res.header('Access-Control-Allow-Origin', origin);
    res.header('Access-Control-Allow-Methods', 'GET,POST,DELETE,OPTIONS,PUT,PATCH');
    res.header('Access-Control-Allow-Headers', 'Content-Type,x-api-key,x-tenant-slug,Authorization');
    res.header('Access-Control-Allow-Credentials', 'true');
  }
  
  if (req.method === 'OPTIONS') {
    res.status(204).send();
    return;
  }
  
  next();
});

// Upload endpoint
app.post('/upload', upload.single('file'), (req, res) => {
  try {
    if (!req.file) {
      return res.status(400).json({ ok: false, error: 'no_file' });
    }

    const { family_slug = 'default', type = 'photo' } = req.body;
    const fileId = uuidv4();
    const filename = req.file.originalname;
    const fileSize = req.file.size;

    const responseData = {
      ok: true,
      id: fileId,
      key: `dev/${family_slug}/${fileId}/${filename}`,
      type: type,
      filename: filename,
      size: fileSize,
      publicUrl: `http://localhost:${port}/uploads/${req.file.filename}`,
      message: 'File uploaded to local development storage'
    };

    console.log(`✅ Mock upload successful: ${filename} (${fileSize} bytes)`);
    res.json(responseData);
  } catch (error) {
    console.error('Upload error:', error);
    res.status(500).json({ ok: false, error: 'upload_failed' });
  }
});

// Serve uploaded files
app.get('/uploads/:filename', (req, res) => {
  const filepath = path.join(uploadDir, req.params.filename);
  if (fs.existsSync(filepath)) {
    res.sendFile(filepath);
  } else {
    res.status(404).send('File not found');
  }
});

// Health check
app.get(['/health', '/status'], (req, res) => {
  res.json({ status: 'ok', message: 'Development mock server' });
});

app.listen(port, () => {
  console.log('🚀 Starting development mock upload server...');
  console.log(`📁 Upload directory: ${uploadDir}`);
  console.log(`🌐 Server running on http://localhost:${port}`);
  console.log('📸 Ready to accept uploads!');
});

module.exports = app;